sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionPlanner.scala

```diff
@@ -99,10 +99,10 @@ private object CosmosPartitionPlanner extends BasicLoggingTrait {
 
     assertOnSparkDriver()
     val lastContinuationTokens: ConcurrentMap[FeedRange, String] = new ConcurrentHashMap[FeedRange, String]()
-    container
-      .getFeedRanges
-      .asScala
-      .flatMapMany(feedRanges => SFlux.fromIterable(feedRanges.asScala))
+
+    ContainerFeedRangesCache
+      .getFeedRanges(container)
```

Comment:
since we are getting from the cache, what will happen if the cache is staled? could it cause less spark partition/job created? missing reading data?

Reply:
Discussed offline, spit logic is handed in ChangeFeed pull model, so no break here.